Client/Frontend/Browser/Tabs/Legacy/LegacyGridTabViewController.swift

```diff
@@ -375,8 +375,10 @@ class LegacyGridTabViewController: UIViewController,
     }
 
     func dismissTabTray() {
-        self.navigationController?.dismiss(animated: true, completion: nil)
-        TelemetryWrapper.recordEvent(category: .action, method: .close, object: .tabTray)
+        DispatchQueue.main.async {
+            self.navigationController?.dismiss(animated: true, completion: nil)
+            TelemetryWrapper.recordEvent(category: .action, method: .close, object: .tabTray)
+        }
```

Comment:
This is not an ideal solution, but since the select tab that happens prior to this is now an async event it is causing crashes on the home page when the view is dismissed since the tab data is not fully resolved after deleting tabs. I'm comfortable putting this solution here because the legacy tab tray is going away soon anyway.